internal/handlers/common/filter.go

```diff
@@ -52,17 +52,17 @@ func FilterDocument(doc, filter *types.Document) (bool, error) {
 func filterDocumentPair(doc *types.Document, filterKey string, filterValue any) (bool, error) {
 	if strings.ContainsRune(filterKey, '.') {
 		// {field1./.../.fieldN: filterValue}
-		path := strings.Split(filterKey, ".")
+		path := types.NewPathFromString(filterKey)
 		// we pass the path without the last key because we want {fieldN: *someValue*}, not just *someValue*
-		docValue, err := doc.GetByPath(path[:len(path)-1]...)
+		docValue, err := doc.GetByPath(path.Remove(path.Len() - 1))
```

Comment:
Do we actually have use cases for methods with indexes instead of just prefixes and suffixes?

I think that line, as well as `path.Element(path.Len() - 1)` below, are complex, and that's something we would like to hide with the good path's API

Reply:
> Do we actually have use cases for methods with indexes?

Some are in `$elemMatch` when it is necessary to delete an item in the middle of an array.

Reply:
Please point me to the line of code

Reply:
https://github.com/FerretDB/FerretDB/pull/551/files#r881425664
It's a path however I don't see how the path that can be used there, it's overcomplication.

Reply:
I'm not sure how that relates to my concern. I was saying that we don't have code that calls `path.Element(x)` with `x` being something other than `0` (prefix) or `path.Len()-1` (suffix). The linked line of code is not about path; the code below is about suffix.

Reply:
Replaced with TrimSuffix/TrimPrefix and Suffix/Prefix methods.